### PR TITLE
Implement poke functionality using diff.Apply function

### PR DESCRIPTION
**Pull request by willhite**

### CODE REVIEW

File: samples/go/downloader/resource_cache.go [hunk 1]
```diff
@@ -22,24 +22,24 @@ type resourceCache struct {
 }
 
 func checkCacheType(c types.Value) (err error) {
-    err = errors.New("resourceCache value is not Map<String, Ref<Blob>>")
-    var m types.Map
-	
-    if err1 := marshal.Unmarshal(c, &m); err1 != nil {
+	err = errors.New("resourceCache value is not Map<String, Ref<Blob>>")
```
Comment (aboodman) [inline samples/go/downloader/resource_cache.go]: Is this an unrelated change?

**Comment by aboodman** [comment]
LGTM

**Comment by willhite** [comment]
jenkins: build this

**Comment by zcstarr** [comment]
jenkins: test this

**Comment by arv** [comment]
jenkins: test this
